dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/BaggageProxy.java

```diff
@@ -19,47 +19,91 @@
 import io.opentelemetry.api.baggage.BaggageBuilder;
 import io.opentelemetry.api.baggage.BaggageEntry;
 
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
 /**
  * This proxy class redirects method calls to Baggage.current(), by doing so it allows people to use @Inject Baggage and get an object which will not become obsolete.
  */
 public class BaggageProxy implements Baggage {
 
+    private static final TraceComponent tc = Tr.register(BaggageProxy.class);
+
     @Override
     public int size() {
-        return Baggage.current().size();
+        try {
+            return Baggage.current().size();
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CEMOT5101.telemetry.error", e));
+            return -1;
```

Comment:
If we can't get the current baggage, the most appropriate thing to do is probably to pretend we have an empty `Baggage`, so I'd return `0` here.